### ISSUE TITLE
A possible false positive in pattern matching with generic type in a different module 

### ISSUE THREAD

**Issue by PowerSnail** [microsoft/pyright#2536]
**Describe the bug**

Pyright seem to report a false positive when a pattern-matching case is a class from a different module, for instance, `case lib.A()`. If `A` is imported into the current namespace (`from lib import A`), however, this doesn't happen.

**To Reproduce**

See below in code example.

**Expected behavior**

No error reported.

**Screenshots or Code**

```python
# lib.py
import typing


T = typing.TypeVar("T")
E = typing.TypeVar("E")

class A(typing.Generic[T]):
    ...


class B(typing.Generic[E]):
    ...


AorB = A[T] | B[E]
```

```python
# error.py
from . import lib

def work(value: lib.A[int] | lib.B[str]):
    match value:
        case lib.A():
            ...
        case lib.B():
            ...
```

In error.py, pyright reports: 

```
/path/to/error.py:5:14 - error: Type of "A" is partially unknown
  Type of "A" is "Type[A[Unknown]]" (reportUnknownMemberType)
/path/to/error.py:7:14 - error: Type of "B" is partially unknown
  Type of "B" is "Type[B[Unknown]]" (reportUnknownMemberType)
```

```python
# no_error.py

from .lib import A, B

def work2(value: A[int] | B[str]):
    match value:
        case A():
            ...
        case B():
            ...
```

No error is reported for no_error.py.

**VS Code extension or command-line**
Both exhibits the same behavior.

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.185, which I just published. It will also be included in the next release of pylance.
